src/core/core.layouts.js

```diff
@@ -3,43 +3,89 @@
 var defaults = require('./core.defaults');
 var helpers = require('../helpers/index');
 
+var extend = helpers.extend;
+
 function filterByPosition(array, position) {
 	return helpers.where(array, function(v) {
-		return v.position === position;
+		return v.pos === position;
 	});
 }
 
 function sortByWeight(array, reverse) {
 	array.forEach(function(v, i) {
-		v._tmpIndex_ = i;
+		v.index = i;
+		v.weight = v.box.weight;
 		return v;
 	});
 	array.sort(function(a, b) {
 		var v0 = reverse ? b : a;
 		var v1 = reverse ? a : b;
 		return v0.weight === v1.weight ?
-			v0._tmpIndex_ - v1._tmpIndex_ :
+			v0.index - v1.index :
 			v0.weight - v1.weight;
 	});
-	array.forEach(function(v) {
-		delete v._tmpIndex_;
-	});
+}
+
+function wrapBoxes(boxes) {
+	var layoutBoxes = [];
+	var i, ilen, box;
+
+	for (i = 0, ilen = (boxes || []).length; i < ilen; ++i) {
+		box = boxes[i];
+		layoutBoxes.push({
+			pos: box.position,
+			box: box
+		});
+	}
+	return layoutBoxes;
+}
+
+function buildLayoutBoxes(boxes) {
+	var layoutBoxes = wrapBoxes(boxes);
+	var left, right, top, bottom, vertical, horizontal;
+
+	left = filterByPosition(layoutBoxes, 'left');
+	right = filterByPosition(layoutBoxes, 'right');
+	top = filterByPosition(layoutBoxes, 'top');
+	bottom = filterByPosition(layoutBoxes, 'bottom');
+
+	// Sort boxes by weight. A higher weight is further away from the chart area
+	sortByWeight(left, true);
+	sortByWeight(right);
+	sortByWeight(top, true);
+	sortByWeight(bottom);
+
+	vertical = left.concat(right);
+	horizontal = top.concat(bottom);
+
+	return {
+		lt: left.concat(top),
```

Comment:
These short names aren't too clear to me. How about calling them `leftAndTop`, `rightAndBottom`, `vertical`, `horizontal`, and `outer`?

Reply:
done